### PR TITLE
General snprintf() hardenings

### PR DESCRIPTION
**Pull request by kbabioch**
This series makes sure that the return code of snprintf() is always evaluated correctly to make sure it does not run into errors silently.

### CODE REVIEW

File: pam_yubico.c [hunk 1]
```diff
@@ -308,7 +308,11 @@ authorize_user_token_ldap (struct cfg *cfg,
       DBG ("Failed allocating %zu bytes", i);
       goto done;
     }
-    sprintf (find, "%s=%s,%s", cfg->user_attr, user, cfg->ldapdn);
+    int j = snprintf (find, i, "%s=%s,%s", cfg->user_attr, user, cfg->ldapdn);
```
Comment (klali) [inline pam_yubico.c]: Please move the variable declaration up to start of the block.

File: util.c [hunk 2]
```diff
@@ -68,7 +68,10 @@ get_user_cfgfile_path(const char *common_path, const char *filename, const struc
     if ((userfile = malloc(len)) == NULL) {
       return 0;
     }
-    snprintf(userfile, len, "%s/%s", common_path, filename);
+    int i = snprintf(userfile, len, "%s/%s", common_path, filename);
```
Comment (klali) [inline util.c]: Same thing here, int i needs to move up.

File: util.c [hunk 3]
```diff
@@ -79,7 +82,10 @@ get_user_cfgfile_path(const char *common_path, const char *filename, const struc
   if ((userfile = malloc(len)) == NULL) {
     return 0;
   }
-  snprintf(userfile, len, "%s/.yubico/%s", user->pw_dir, filename);
+  int i = snprintf(userfile, len, "%s/.yubico/%s", user->pw_dir, filename);
```
Comment (klali) [inline util.c]: Here as well, please move the declaration up.

File: util.c [hunk 4]
```diff
@@ -332,7 +338,14 @@ check_user_challenge_file(const char *chalresp_path, const struct passwd *user,
     ret = AUTH_ERROR;
     goto out;
   }
-  snprintf(userfile_pattern, len, "%s-*", userfile);
+
+  int i = snprintf(userfile_pattern, len, "%s-*", userfile);
```
Comment (klali) [inline util.c]: And here.

**Comment by kbabioch** [comment]
Here you go. Shouldn't this be something that would be detected by Travis?

**Comment by klali** [comment]
Yes, this should ideally be detected. I compile it with --enable-gcc-warnings to configure, unfortunately that generates noise/things we/I haven't cared about earlier. I try to make sure that changes don't introduce new warnings in that.

Looks good, merging.

**Pull request merged**